src/lib.rs

```diff
@@ -389,6 +392,35 @@ impl Hunk {
     pub fn is_pure_removal(&self) -> bool {
         self.after.is_empty()
     }
+
+    /// Performs a byte-diff of the hunk
+    pub fn byte_diff(&self, diff: &mut Diff, input: &InternedInput<&str>) {
+        let Hunk { before, after } = self.clone();
+        let before_bytes: &[u8] = &before
+            .flat_map(|token| input.interner[input.before[token as usize]].bytes())
+            .collect::<Vec<_>>();
+        let after_bytes: &[u8] = &after
+            .flat_map(|token| input.interner[input.after[token as usize]].bytes())
+            .collect::<Vec<_>>();
+        diff.removed.clear();
+        diff.removed.resize(before_bytes.len(), false);
+        diff.added.clear();
+        diff.added.resize(after_bytes.len(), false);
+        if self.is_pure_removal() {
+            diff.removed.fill(true);
+        } else if self.is_pure_insertion() {
+            diff.added.fill(true);
+        } else {
+            let input = InternedInput::new(bytes(before_bytes), bytes(after_bytes));
+            diff.compute_with(
+                Algorithm::Myers,
```

Comment:
Point 5

Reply:
I am specifically referring to https://github.com/pascalkuthe/imara-diff/blob/8de65af4621b4d83c35b458a73f544cb68842edb/src/lib.rs#L221

Reply:
I suspect that word diffs are rarely large enough for that heuristic to kick in and if it does then you probably don't care about the results anyway (way bigger than a human could read). In general the minimal node is more aimed at usecases where you are not producing a readable diff but rather something for storage/network transfers (possibly not even riffing text). Word diffs are only really useful for text based diffs and probably usually human readable ones so I suspect it wouldn't be commonly used. 

With these kind of things I tend to take a eait and see approach: I would hardcode the algorithm until a real world usecase for other algorithms actually materializes